src/WebJobs.Script.WebHost/Security/KeyManagement/SecretManager.cs

```diff
@@ -90,12 +89,26 @@ public async virtual Task<HostSecretsInfo> GetHostSecretsAsync()
                         _logger.LogDebug("Loading host secrets");
 
                         hostSecrets = await LoadSecretsAsync<HostSecrets>();
-                        if (hostSecrets == null)
+                        try
                         {
-                            // host secrets do not yet exist so generate them
-                            _logger.LogDebug(Resources.TraceHostSecretGeneration);
-                            hostSecrets = GenerateHostSecrets();
-                            await PersistSecretsAsync(hostSecrets);
+                            if (hostSecrets == null)
+                            {
+                                // host secrets do not yet exist so generate them
+                                _logger.LogDebug(Resources.TraceHostSecretGeneration);
+                                hostSecrets = GenerateHostSecrets();
+                                await PersistSecretsAsync(hostSecrets);
+                            }
+                        }
+                        catch (Exception ex)
+                        {
+                            _logger.LogDebug("Potential conflict, attempting to load host secrets for the second time.");
```

Comment:
Couple comments on this:
- Let's pass the exception in here so we have it captured. `_logger.LogDebug(ex, "....")`
- Reword a little so it's clear what happened: `Exception while generating host secrets. This can happen if another instance is also generating secrets. Attempting to read secrets again.`